ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerThread.java

```diff
@@ -175,6 +205,101 @@ protected void advanceToken(RemoteReplicaInfo remoteReplicaInfo, ExchangeMetadat
     remoteReplicaInfo.setLocalLagFromRemoteInBytes(exchangeMetadataResponse.localLagFromRemoteInBytes);
     // reset stored metadata response for this replica so that we send next request for metadata
     remoteReplicaInfo.setExchangeMetadataResponse(new ExchangeMetadataResponse(ServerErrorCode.No_Error));
+    logReplicationStatus(remoteReplicaInfo, exchangeMetadataResponse);
+  }
+
+  /**
+   * Write to a given file at a given offset
+   * @param seekableByteChannel File descriptor the file to append to
+   * @param text Text to append
+   * @param offset Offset to write to in the file
+   * @return True if append was successful, false otherwise
+   */
+  protected boolean writeToFile(SeekableByteChannel seekableByteChannel, String text, int offset) {
+    try {
+      if (offset >= 0) {
+        seekableByteChannel.position(offset);
+      }
+      seekableByteChannel.write(ByteBuffer.wrap(text.getBytes(StandardCharsets.UTF_8)));
+      return true;
+    } catch (IOException e) {
+      logger.error(e.toString());
+      return false;
+    }
+  }
+
+  /**
+   * Appends to a given file
+   * @param seekableByteChannel File descriptor the file to append to
+   * @param text Text to append
+   * @return True if append was successful, false otherwise
+   */
+  protected boolean appendToFile(SeekableByteChannel seekableByteChannel, String text) {
+    return writeToFile(seekableByteChannel, text, -1);
+  }
+
+  /**
+   * Creates or returns an existing file descriptor for file replicaCheckStatus
+   * @param remoteReplicaInfo The remote replica that is being replicated from
+   * @return Return a file descriptor of type SeekableByteChannel
+   */
+  protected SeekableByteChannel getOrCreateStatusFileFd(RemoteReplicaInfo remoteReplicaInfo) {
+    String filePath = basePath
+        + File.separator + remoteReplicaInfo.getReplicaId().getPartitionId().getId()
+        + File.separator + remoteReplicaInfo.getReplicaId().getDataNodeId().getHostname()
+        + File.separator + "replicaCheckStatus";
+    /**
+     * NOTE: Working with these file options has been a pain. They do not work as advertised in documentation.
```

Comment:
Can we remove this comment? As much as I sympathize with the pain, this isn't useful for the reader.

Reply:
I'd like to have this for the record, at least for myself if not someone else. I spent close to half the day figuring out why the code wasn't working as expected and I don't want to repeat it. I am sure it will be useful to someone other than me. At the very least, it's not misleading.

Reply:
We can add comment to mention the flags needed. It should be enough, especially it's OSS.

Reply:
Yeah, certain combos of the flags work and some don't.

Reply:
Again, atleast because its open source code, lets not make any subjective statements about APIs that aren't part of this code base. Objective statements like "ABC flags don't work as expected in library X with version Y" are ok.